daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc.hs

```diff
@@ -208,7 +222,152 @@ execCompile inputFile outputFile opts = withProjectRoot $ \relativize -> do
+        Right (name, main, exposedModules, version, dependencies)
+
+    mkConfFile ::
+           String
+        -> String
+        -> LF.Version
+        -> [String]
+        -> [FilePath]
+        -> (String, B.ByteString)
+    mkConfFile name version lfVersion exposedMods deps = (confName, bs)
+      where
+        confName = name ++ ".conf"
+        lfVersionStr = lfVersionString lfVersion
+        bs =
+            BSC.pack $
+            unlines
+                [ "name: " ++ name
+                , "id: " ++ name
+                , "key: " ++ name
+                , "version: " ++ version
+                , "exposed: True"
+                , "exposed-modules: " ++ unwords exposedMods
+                , "import-dirs: ${pkgroot}" </> lfVersionStr </> name
+                , "library-dirs: ${pkgroot}" </> lfVersionStr </> name
+                , "data-dir: ${pkgroot}" </> lfVersionStr </> name
+                , "depends: " ++
+                  unwords [dropExtension $ takeFileName dep | dep <- deps]
+                ]
+
+    -- The default output filename is based on Maven coordinates if
+    -- the package name is specified via them, otherwise we use the
+    -- name.
+    defaultDarFile name =
+        case Split.splitOn ":" name of
+            [_g, a, v] -> a <> "-" <> v <> ".dar"
+            _otherwise -> name <> ".dar"
+    targetFilePath name = fromMaybe (defaultDarFile name) mbOutFile
+
+-- | Create the project package database containing the given dar packages.
+createProjectPackageDb :: LF.Version -> [FilePath] -> IO ()
+createProjectPackageDb lfVersion fps = do
+    let dbPath = projectPackageDatabase </> lfVersionString lfVersion
+    createDirectoryIfMissing True dbPath
+    let fps0 = filter (`notElem` basePackages) fps
+    forM_ fps0 $ \fp -> do
+        bs <- BSL.readFile fp
+        let pkgName = takeBaseName fp
+        let archive = toArchive bs
+        let confFiles =
+                [ e
+                | e <- zEntries archive
+                , ".conf" `isExtensionOf` eRelativePath e
+                ]
+        let dalfs =
+                [ e
+                | e <- zEntries archive
+                , ".dalf" `isExtensionOf` eRelativePath e
+                ]
+        let srcs =
+                [ e
+                | e <- zEntries archive
+                , pkgName `isPrefixOf` eRelativePath e
+                ]
+        forM_ dalfs $ \dalf ->
+            BSL.writeFile (dbPath </> eRelativePath dalf) (fromEntry dalf)
+        forM_ confFiles $ \conf ->
+            BSL.writeFile
+                (dbPath </> (takeFileName $ eRelativePath conf))
+                (fromEntry conf)
+        createDirectoryIfMissing True $ dbPath </> pkgName
+        forM_ srcs $ \src ->
+            BSL.writeFile (dbPath </> eRelativePath src) (fromEntry src)
+    sdkRoot <- getSdkPath
+    callCommand $
+        unwords
+            [ sdkRoot </> "damlc/resources/ghc-pkg"
```

Comment:
This should probably use `locateRunfiles` instead so it works with `bazel run`, ghci and the final distribution.

Reply:
ah good idea!